drivers/flash/CMakeLists.txt

```diff
@@ -56,6 +55,15 @@ if(CONFIG_SOC_FLASH_STM32)
   endif()
 endif()
 
+if(CONFIG_SOC_FLASH_NRF)
+  zephyr_library_sources(soc_flash_nrf.c)
```

Comment:
move of zephyr_library_sources for nrf driver seams to be unnecessary. I would keep it as it was.